extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -467,10 +515,25 @@ public boolean canRestore()
   public void stopGracefully()
```

Comment:
I can swap the implementations of stopGracefully and stopAbruptly since the abrupt one does what you described.